casanovo/casanovo.py

```
@@ -521,183 +462,205 @@ def setup_logging(
+def _normalize(s: str) -> str:
+    return re.sub(r"[-_\s]", "", s).lower()
+
+
+def _parse_ckpt(filename: str) -> Optional[Tuple[str, Tuple[int, int, int]]]:
+    m = _CKPT_RE.match(os.path.basename(filename))
+    if not m:
+        return None
+    return m.group(1), (int(m.group(2)), int(m.group(3)), int(m.group(4)))
+
+
+def _resolve_selector(selector: str, candidates: List[str]) -> str:
```

Ruff (0.15.14)

[error] Undefined name `List`

(F821)